snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -357,6 +364,53 @@ def cleanup(self):
                 if os.path.splitext(f)[1] == ".desktop":
                     os.remove(os.path.join(self.meta_gui_dir, f))
 
+    def _finalize_icon(self) -> Optional[pathlib.Path]:
+        """Make sure icon is properly configured, fetching from a remote
```

Comment:
This does not fit a proper docstring single-line statement.

Description can follow in the paragraph below

Reply:
fixed